packages/react/src/useAction.ts

```diff
@@ -79,9 +79,43 @@ export const useAction = <
     transformedResult,
     useCallback(
       async (variables, context) => {
+        if (action.hasAmbiguousIdentifier) {
+          if (Object.keys(variables).some((key) => !action.paramOnlyVariables?.includes(key) && key !== action.modelApiIdentifier)) {
+            throw Error(`Invalid arguments found in variables. Did you mean to use ({ ${action.modelApiIdentifier}: { ... } })?`);
+          }
+        }
+
+        let newVariables: Exclude<F["variablesType"], null | undefined>;
+        if (action.hasCreateOrUpdateEffect) {
+          if (
+            action.modelApiIdentifier in variables &&
+            typeof variables[action.modelApiIdentifier] === "object" &&
+            variables[action.modelApiIdentifier] !== null
+          ) {
+            newVariables = variables;
+          } else {
+            newVariables = {
+              [action.modelApiIdentifier]: {},
+            } as Exclude<F["variablesType"], null | undefined>;
+            for (const [key, value] of Object.entries(variables)) {
+              if (action.paramOnlyVariables?.includes(key)) {
+                newVariables[key] = value;
+              } else {
+                if (key === "id") {
```

Comment:
`id` is kind of special in that if it's an update effect then we have to include it as a separate variable

Reply:
👍 

Should we be checking if it's an update? Or can we look at `action.variables` to see if it should be nested under the model API identifier or not? I'm just wondering if we could future-proof against creates where the end user can supply an id, but it won't be "lifted" to a variable like we do for updates.

Reply:
Changed this up to check `action.variables`